src/framework/dispatchers/RequestDispatcher.kt

```diff
@@ -42,6 +43,8 @@ open class RequestDispatcher: Dispatcher<ApiGatewayRequest, Any> {
 
             if(result.result == SOAResultType.SUCCESS) {
                 return result.data
+            } else if (result.message == "Not Found") {
```

Comment:
this check wont be needed if you make the change to the controller to throw the error